pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.utils;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+
+public class LLCSegmentNameHolder extends  SegmentNameHolder implements Comparable {
+  private final static String DATE_FORMAT = "yyyyMMdd'T'HHmm'Z'";
+  private final String _tableName;
+  private final int _partitionId;
+  private final int _sequenceNumber;
+  private final String _creationTime;
+  private final String _segmentName;
+
+  public LLCSegmentNameHolder(String segmentName) {
+    String[] parts =  segmentName.split("__");
+    if (parts.length != 4) {
+      throw new RuntimeException(segmentName + " is not a Low level consumer segment name");
+    }
+    _segmentName = segmentName;
+    _tableName = parts[0];
+    _partitionId = Integer.valueOf(parts[1]);
+    _sequenceNumber = Integer.valueOf(parts[2]);
+    _creationTime = parts[3];
+  }
+
+  public LLCSegmentNameHolder(String tableName, int partitionId, int sequenceNumber, long msSinceEpoch) {
+    if (!isValidComponentName(tableName)) {
+      throw new RuntimeException("Invalid table name " + tableName);
+    }
+    _tableName = tableName;
+    _partitionId = partitionId;
+    _sequenceNumber = sequenceNumber;
+    // ISO8601 date: 20160120T1234Z
+    DateTime dateTime = new DateTime(msSinceEpoch, DateTimeZone.UTC);
+    _creationTime = dateTime.toString("yyyyMMdd'T'HHmm'Z'");
+    _segmentName = tableName + SEPARATOR + partitionId + SEPARATOR + sequenceNumber + SEPARATOR + _creationTime;
+  }
+
+  public String getTableName() {
+    return _tableName;
+  }
+
+  public int getPartitionId() {
+    return _partitionId;
+  }
+
+  public int getSequenceNumber() {
+    return _sequenceNumber;
+  }
+
+  public String getCreationTime() {
+    return _creationTime;
+  }
+
+  public String getSegmentName() {
+    return _segmentName;
+  }
+
+  @Override
+  public int compareTo(Object o) {
+    LLCSegmentNameHolder other = (LLCSegmentNameHolder)o;
```

Comment:
See SegmentNameBuilderTest.LLCHolderTest().

Does that suffice, or did you have more tests in mind?